src/Frame.cpp

```diff
@@ -1,227 +1,493 @@
+            }
+            if (xCell < (xCells - 1) && yCell > 0) {
+               nbList[cellIndex].push_back(cellIndex - xCells + 1);
+            }
+            if (xCell > 0 && zCell < (zCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xyTotalCells - 1);
+            }
+            if (yCell > 0 && zCell < (zCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xyTotalCells - xCells);
+            }
+            if (xCell < (xCells - 1) && yCell < (yCells - 1) && zCell < (zCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xyTotalCells + xCells + 1);
+            }
+            if (xCell > 0 && yCell < (yCells - 1) && zCell < (zCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xyTotalCells + xCells - 1);
+            }
+            if (xCell < (xCells - 1) && yCell > 0 && zCell < (zCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xyTotalCells - xCells + 1);
+            }
+            if (xCell > 0 && yCell > 0 && zCell < (zCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xyTotalCells - xCells - 1);
+            }
+            if (xCell) {
+               nbList[cellIndex].push_back(cellIndex - 1);
+            }
+            if (yCell) {
+               nbList[cellIndex].push_back(cellIndex - xCells);
+            }
+            if (zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells);
+            }
+            if (xCell && yCell) {
+               nbList[cellIndex].push_back(cellIndex - xCells - 1);
+            }
+            if (xCell && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells - 1);
+            }
+            if (yCell && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells - xCells);
+            }
+            if (xCell && yCell < (yCells - 1)) {
+               nbList[cellIndex].push_back(cellIndex + xCells - 1);
+            }
+            if (xCell < (xCells - 1) && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells + 1);
+            }
+            if (yCell < (yCells - 1) && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells + xCells);
+            }
+            if (xCell && yCell && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells - xCells - 1);
+            }
+            if (xCell < (xCells - 1) && yCell && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells - xCells + 1);
+            }
+            if (xCell && yCell < (yCells - 1) && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells + xCells - 1);
+            }
+            if (xCell < (xCells - 1) && yCell < (yCells - 1) && zCell) {
+               nbList[cellIndex].push_back(cellIndex - xyTotalCells + xCells + 1);
+            }
+
+            //End of the list
+            nbList[cellIndex].push_back(-1);
```

Comment:
this is not needed for std::vector, since we already know where is the end of a sub-vector.

Reply:
Absolutely right. It was needed for the array used in the previous version but not anymore with std::vector. Thanks.